src/sage/rings/semirings/tropical_polynomial.py

```diff
@@ -0,0 +1,983 @@
+
+        OUTPUT:
+
+        If ``xmin`` and ``xmax`` is given, then it return a plot of
+        piecewise linear function of ``self`` with the axes start from
+        ``xmin`` to ``xmax``. Otherwise, the domain will start from the
+        the minimum root of ``self`` minus 1 to maximum root of ``self``
+        plus 1. If the function of ``self`` is constant or linear, then
+        the default domain will be `[-1,1]`.
+
+        EXAMPLES:
+
+        If the tropical semiring use a max-plus algebra, then the graph
+        will be of piecewise linear convex function::
+
+            sage: T = TropicalSemiring(QQ, use_min=False)
+            sage: R.<x> = PolynomialRing(T)
+            sage: p1 = R([4,2,1,3]); p1
+            3*x^3 + 1*x^2 + 2*x + 4
+            sage: p1.roots()
+            [1/3, 1/3, 1/3]
+            sage: p1.plot()
+            Graphics object consisting of 1 graphics primitive
+        
+        .. PLOT::
+            :width: 300 px
+
+            T = TropicalSemiring(QQ, use_min=False)
+            R = PolynomialRing(T, 'x')
+            p1 = p1 = R([4,2,1,3])
+            sphinx_plot(p1.plot())
+        
+        A different result will be obtained if the tropical semiring employs
+        a min-plus algebra. Rather, a graph of the piecewise linear concave
+        function will be obtained::
+
+            sage: T = TropicalSemiring(QQ, use_min=True)
+            sage: R.<x> = PolynomialRing(T)
+            sage: p1 = R([4,2,1,3])
+            sage: p1.roots()
+            [-2, 1, 2]
+            sage: p1.plot()
+            Graphics object consisting of 1 graphics primitive
+        
+        .. PLOT::
+            :width: 300 px
+
+            T = TropicalSemiring(QQ, use_min=True)
+            R = PolynomialRing(T, 'x')
+            p1 = R([4,2,1,3])
+            sphinx_plot(plot(p1, xmin=-4, xmax=4))
+        
+        TESTS:
+
+        If ``xmin`` or ``xmax`` is given as an input, then the others also
+        have to be given. Otherwise it will raise an error::
+
+            sage: plot(p1, 5)
+            Traceback (most recent call last):
+            ...
+            ValueError: Expected 2 inputs for xmin and xmax, but got 1
+        
+        Error also occured when ``xmin`` is greater or equal than``xmax``::
+
+            sage: plot(p1, 5, 3)
+            Traceback (most recent call last):
+            ...
+            ValueError: xmin = 5 should be less than xmax = 3
+        """
+        from sage.plot.plot import plot
+        f = self.piecewise_function()
+        if xmin is None and xmax is None:
+            roots = sorted(self.roots())
+            if (not roots) or self.parent().base().zero() in roots:
+                return plot(f, xmin=-1, xmax=1)
+            else:
+                return plot(f, xmin=roots[0]-1, xmax=roots[-1]+1)
+        elif xmin is None or xmax is None:
+            raise ValueError("Expected 2 inputs for xmin and xmax, but got 1")
```

Comment:
```suggestion
            raise ValueError("expected 2 inputs for xmin and xmax, but got 1")
```
Python convention that we follow.